### ISSUE TITLE
Longer strings are not sorted naturally

### ISSUE THREAD

**Issue by herwinw** [rwz/natural_sort#2]
```
irb(main):008:0> NaturalSort.sort(['0.0.2', '0.0.10', '0.0.0.2', '0.0.0.10'])
=> ["0.0.0.10", "0.0.0.2", "0.0.2", "0.0.10"]
irb(main):009:0> NaturalSort::VERSION
=> "0.2.0"
```

The tuple `('0.0.2', '0.0.10')` is sorted as expected, but when adding an extra digit the sorting becomes a regular string sort

**Comment by herwinw** [comment]
After trying some more: the bug occurs when sortings strings with only number and dots, when the number of numbers is even:

```
require 'irb(main):001:0> require 'natural_sort'
=> true
irb(main):002:0> NaturalSort.sort(%w[1.1 1.2 1.10])
=> ["1.1", "1.10", "1.2"]
irb(main):003:0> NaturalSort.sort(%w[1.2.1 1.2.2 1.2.10])
=> ["1.2.1", "1.2.2", "1.2.10"]
irb(main):004:0> NaturalSort.sort(%w[1.2.1 1.2.2 1.2.10])
=> ["1.2.1", "1.2.2", "1.2.10"]
irb(main):005:0> NaturalSort.sort(%w[1.2.3.1 1.2.3.2 1.2.3.10])
=> ["1.2.3.1", "1.2.3.10", "1.2.3.2"]
irb(main):006:0> NaturalSort.sort(%w[1.2.3.4.1 1.2.3.4.2 1.2.3.4.10])
=> ["1.2.3.4.1", "1.2.3.4.2", "1.2.3.4.10"]
```

This is due to the tokenizer:

```ruby
class SegmentedString
  TOKENIZER = /\d+(?:\.\d+)?|\D/

  def tokens
    @tokens ||= input.scan(TOKENIZER)
  end
end
```

Which results in sets of 2 integer:

```
irb(main):007:0> '1.2.3.4.1'.scan(/\d+(?:\.\d+)?|\D/)
=> ["1.2", ".", "3.4", ".", "1"]
```

**Comment by rwz** [comment]
Thanks for reporting this. I think I've addressed this in the commits referenced above. The fix also changed the behavior in a way that affects some of the existing specs. Since there's no "reference" for natural sorting that I'm aware of, and the new behavior still looks "natural" enough, I'm gonna call it a day and declare this to be "fixed"

**Comment by rwz** [comment]
Released version 0.3.0
